apps/desktop-wallet/src/api/addressQueries.ts

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { AddressHash, client } from '@alephium/shared'
+import { queryOptions } from '@tanstack/react-query'
+
+interface AddressBalanceQueryProps {
+  addressHash: AddressHash
+  networkId: number
+  latestTxHash?: string
+}
+
+// Adding latestTxHash in queryKey ensures that we'll refetch when new txs arrive.
+// Adding networkId in queryKey ensures that switching the network we get different data.
+// TODO: Should we add explorerBackendUrl instead?
+export const addressBalanceQuery = ({ addressHash, networkId, latestTxHash }: AddressBalanceQueryProps) =>
+  queryOptions({
+    queryKey: ['address', 'balance', { addressHash, latestTxHash, networkId }],
+    queryFn: () => client.explorer.addresses.getAddressesAddressBalance(addressHash),
+    staleTime: Infinity
+  })
```

Comment:
> calling this query multiple times will not trigger a call to the server until gcTime is reached and data is garbage collected

This is incorrect. Since the query is `staleTime: Infinity` it means that it's always "Fresh". That means it will never get garbage collected. Only "Stale" queries get garbage collected.